### PR TITLE
PAPP-24320: Add a new optional field "Audit comment"

### PR DESCRIPTION
**Pull request by paulnawat**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Add a new optional field Audit comment to use with a Policy rule

## What is the current behavior? (OPTIONAL)
- Policy rule updates cannot have any attached comments

## What is the new behavior? (OPTIONAL)
- Add a new optional field Audit comment to use with a Policy rule

---
Thanks for contributing!

### CODE REVIEW

**Review by gary-phantom — APPROVED** [review]

**Comment by paulnawat** [comment]
The integration tests failed here, because the updated tests have not been merged to next yet.
Green pipeline: pipelines/4760834

**Review by alexa-phantom — APPROVED** [review]

File: panorama_connector.py [hunk 1]
```diff
@@ -1580,19 +1647,26 @@ def _run_query(self, param):
...
 
         if device_group.lower() == PAN_DEV_GRP_SHARED:
             return '/config/shared'
 
-        return "/config/devices/entry/device-group/entry[@name='{device_group}']".format(device_group=device_group)
+        formatted_device_entry_name = ''
+        if device_entry_name:
+            formatted_device_entry_name = "[@name='{}']".format(device_entry_name)
+
+        return "/config/devices/entry{formatted_device_entry_name}/device-group/entry[@name='{device_group}']".format(
```
Comment (rgil-splunk) [inline panorama_connector.py]: can we move this big string to a constant?

File: panorama_connector.py [hunk 2]
```diff
@@ -1580,19 +1647,26 @@ def _run_query(self, param):
...
-    def _get_config_xpath(self, param):
+    def _get_config_xpath(self, param, device_entry_name=''):
+        """Return the xpath to the specified device group"""
 
+        if device_entry_name:
+            self.debug_print('Getting the Config xpath for the device entry name %s' % device_entry_name)
         device_group = self._handle_py_ver_compat_for_input_str(param[PAN_JSON_DEVICE_GRP])
 
         if device_group.lower() == PAN_DEV_GRP_SHARED:
             return '/config/shared'
```
Comment (rgil-splunk) [inline panorama_connector.py:1616]: ditto

**Review by rgil-splunk — APPROVED** [review]

File: panorama_connector.py (hunk 1 above)
Comment (paulnawat) [reply panorama_connector.py]: Seeing it's being used in only one place, I feel putting it here can make it easy to access (i.e. we don't need to jump to another file/location to read its value. What do you think?

Comment (paulnawat) [reply panorama_connector.py]: As discussed, convert the long string to `DEVICE_GRP_XPATH`

**Pull request merged**